packages/core/testing/src/r3_test_bed.ts

```diff
@@ -383,6 +402,14 @@ export class TestBedRender3 implements TestBed {
   }
 
   createComponent<T>(type: Type<T>): ComponentFixture<T> {
+    // Check if a configuration has been provided to throw when an unknown element is found during
+    // the template compilation
+    const instanceOptions = this._instanceTemplateCompilationOptions;
+    const environmentOptions = TestBedRender3._environmentTemplateCompilationOptions;
+    let shouldThrowOnUnknownElements = instanceOptions?.throwOnUnknownElement ??
+        environmentOptions?.throwOnUnknownElement ?? THROW_ON_UNKNOWN_ELEMENTS_DEFAULT;
+    setShouldThrowErrorOnUnknownElement(shouldThrowOnUnknownElements);
```

Comment:
It'd be great to move the logic that calculates an actual state to a separate function (similar to `shouldRethrowTeardownErrors`) and use it here:
```suggestion
    setUnknownElementStrictMode(this.shouldThrowOnUnknownElements());
```